Fix sndio support

lxeiqr opened the pull request:
This patch allows sandboxed programs to access a sndiod server and play audio on systems where sndio handles audio.

SkewedZeppelin reviewed (changes requested):
-1, don't add /tmp to this, will effectively apply private-tmp to all profiles using whitelist-common

SkewedZeppelin commented:
This is likely something you should add to your locals.

rusty-snake reviewed (changes requested):
- [ ] Move it into the common block (the first one)
- [x] Only `whitelist` in `${HOME}` (if `whitelist /tmp/sndio` is necessary, we should add it to `private-tmp`)

lxeiqr commented:
>     * [ ]  Move it into the common block (the first one)
> 
>     * [ ]  Only `whitelist` in `${HOME}` (if `whitelist /tmp/sndio` is necessary, we should add it to `private-tmp`)
I've removed that last line and moved the cookie whitelist into the common block. However, I'm not quite sure how to handle the /tmp/sndio directory. Usually it contains UNIX domain sockets (e.g. /tmp/sndio/sndio0) used to connect to the server.

rusty-snake commented on etc/inc/whitelist-common.inc:
```diff
@@ -24,6 +24,7 @@ whitelist ${HOME}/.local/share/icons
 whitelist ${HOME}/.local/share/mime
 whitelist ${HOME}/.mime.types
 whitelist ${HOME}/.uim.d
+whitelist ${HOME}/.sndio/cookie
```
```suggestion
whitelist ${HOME}/.sndio/cookie
whitelist ${HOME}/.uim.d
```

rusty-snake commented:
> However, I'm not quite sure how to handle the /tmp/sndio directory. Usually it contains UNIX domain sockets (e.g. /tmp/sndio/sndio0) used to connect to the server.

I think a `profile_add("whitelist /tmp/sndio");` in `fs_private_tmp` works

https://github.com/netblue30/firejail/blob/447e30e2c9b24f4ec6af5520a37a77068e66383b/src/firejail/fs.c#L1251-L1299

lxeiqr commented:
> > However, I'm not quite sure how to handle the /tmp/sndio directory. Usually it contains UNIX domain sockets (e.g. /tmp/sndio/sndio0) used to connect to the server.
> 
> I think a `profile_add("whitelist /tmp/sndio");` in `fs_private_tmp` works
> 
> https://github.com/netblue30/firejail/blob/447e30e2c9b24f4ec6af5520a37a77068e66383b/src/firejail/fs.c#L1251-L1299

Done, now it works properly without any extra tweaking.

lxeiqr commented:
Done, now it works properly without any extra tweaking.

rusty-snake commented on src/firejail/fs.c:
```diff
@@ -1274,6 +1274,9 @@ void fs_private_tmp(void) {
 	// read-only x11 directory
 	profile_add("read-only /tmp/.X11-unix");
 
+        // whitelist sndio directory
+        profile_add("whitelist /tmp/sndio");
```
8 spaces -> 1 tab

rusty-snake reviewed (commented):
LGTM

netblue30 commented:
Thanks for the patch!
